fortitude/src/options.rs

```diff
@@ -0,0 +1,287 @@
+        example = r#"ignore = ["superfluous-implicit-none"]"#
+    )]
+    pub ignore: Option<Vec<RuleSelector>>,
+
+    // TODO: fix default and example when default rules decided!
+    /// A list of rule codes or prefixes to enable. Prefixes can specify exact
+    /// rules (like `T003` or `superfluous-implicit-none`), entire categories
+    /// (like `T` or `typing`), or anything in between.
+    ///
+    /// When breaking ties between enabled and disabled rules (via `select` and
+    /// `ignore`, respectively), more specific prefixes override less
+    /// specific prefixes.
+    #[option(
+        default = r#"["E4", "E7", "E9", "F"]"#,
+        value_type = "list[RuleSelector]",
+        example = r#"
+            # On top of the defaults (`E4`, E7`, `E9`, and `F`), enable flake8-bugbear (`B`) and flake8-quotes (`Q`).
+            select = ["E4", "E7", "E9", "F", "B", "Q"]
+        "#
+    )]
+    pub select: Option<Vec<RuleSelector>>,
+
+    // TODO: fix default and example when default rules decided!
+    /// A list of rule codes or prefixes to enable, in addition to those
+    /// specified by [`select`](#check_select).
+    #[option(
+        default = "[]",
+        value_type = "list[RuleSelector]",
+        example = r#"
+            # On top of the default `select` (`E4`, E7`, `E9`, and `F`), enable flake8-bugbear (`B`) and flake8-quotes (`Q`).
+            extend-select = ["B", "Q"]
+        "#
+    )]
+    pub extend_select: Option<Vec<RuleSelector>>,
+
+    // File resolver options
+    /// A list of file extensions to check
+    #[option(
+        default = r#"["f90", "F90", "f95", "F95", "f03", "F03", "f08", "F08", "f18", "F18", "f23", "F23"]"#,
+        value_type = "list[str]",
+        example = r#"["f90", "fpp"]"#
+    )]
+    pub file_extensions: Option<Vec<String>>,
+
+    /// A list of file patterns to exclude from formatting and linting.
+    ///
+    /// Exclusions are based on globs, and can be either:
+    ///
+    /// - Single-path patterns, like `.mypy_cache` (to exclude any directory
+    ///   named `.mypy_cache` in the tree), `foo.f90` (to exclude any file named
+    ///   `foo.f90`), or `foo_*.f90` (to exclude any file matching `foo_*.f90` ).
+    /// - Relative patterns, like `directory/foo.f90` (to exclude that specific
+    ///   file) or `directory/*.f90` (to exclude any Fortran files in
+    ///   `directory`). Note that these paths are relative to the project root
+    ///   (e.g., the directory containing your `fpm.toml`).
+    ///
+    /// For more information on the glob syntax, refer to the [`globset` documentation](https://docs.rs/globset/latest/globset/#syntax).
+    ///
+    /// Note that you'll typically want to use
+    /// [`extend-exclude`](#extend-exclude) to modify the excluded paths.
+    #[option(
+        default = r#"[".git", ".git-rewrite", ".hg", ".svn", "venv", ".venv", "pyenv", ".pyenv", ".eggs", "site-packages", ".vscode", "build", "_build", "dist", "_dist"]"#,
```

Comment:
Is this default needed? This list of default excluded paths is mirrored in `fs.rs`, but there it goes one further, as they're treated as a default that can't be overridden. Setting `exclude` here wouldn't override the stated default behaviour as it would for other options.

Reply:
I'm like 95% certain the default here is _just_ for the docs. The `option` macro requires a `default` value, so we do have to supply something.

Reply:
Ah, I see. I think my brain had automatically applied the filter that `#[option(...)]` was a `serde` thing.